controllers/components/dashboard/dashboard_support.go

```diff
@@ -101,3 +101,147 @@ func computeComponentName() string {
 func GetAdminGroup() string {
 	return adminGroups[cluster.GetRelease().Name]
 }
+
+// TODO: to be removed: https://issues.redhat.com/browse/RHOAIENG-21080
+func updateSpecFields(obj *unstructured.Unstructured, updates map[string][]any) (bool, error) {
+	updated := false
+
+	for field, newData := range updates {
+		existingField, exists, err := unstructured.NestedSlice(obj.Object, "spec", field)
+		if err != nil {
+			return false, fmt.Errorf("failed to get field '%s': %w", field, err)
+		}
+
+		if !exists || len(existingField) == 0 {
+			if err := unstructured.SetNestedSlice(obj.Object, newData, "spec", field); err != nil {
+				return false, fmt.Errorf("failed to set field '%s': %w", field, err)
+			}
+			updated = true
+		}
+	}
+
+	return updated, nil
+}
+
+// TODO: to be removed: https://issues.redhat.com/browse/RHOAIENG-21080
+func getNotebookSizesData() []any {
+	return []any{
+		map[string]any{
+			"name": "Small",
+			"resources": map[string]any{
+				"requests": map[string]any{
+					"cpu":    "1",
+					"memory": "4Gi",
+				},
+				"limits": map[string]any{
+					"cpu":    "3",
+					"memory": "8Gi",
+				},
+			},
+		},
+		map[string]any{
+			"name": "Small",
+			"resources": map[string]any{
+				"requests": map[string]any{
+					"cpu":    "1",
+					"memory": "8Gi",
+				},
+				"limits": map[string]any{
+					"cpu":    "2",
+					"memory": "8Gi",
+				},
+			},
+		},
```

Comment:
Is there supposed to be 2 `Small` blocks for the notebook size configuration?

Reply:
oops my bad, will fix that